In [2]:
import pandas as pd
import numpy as np

filename = pd.read_excel("titanic3.xls")
df = pd.DataFrame(filename)

In [3]:
df.head()

pclass  survived                                             name     sex  \
0       1         1                    Allen, Miss. Elisabeth Walton  female   
1       1         1                   Allison, Master. Hudson Trevor    male   
2       1         0                     Allison, Miss. Helen Loraine  female   
3       1         0             Allison, Mr. Hudson Joshua Creighton    male   
4       1         0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  female   

       age  sibsp  parch  ticket      fare    cabin embarked boat   body  \
0  29.0000      0      0   24160  211.3375       B5        S    2    NaN   
1   0.9167      1      2  113781  151.5500  C22 C26        S   11    NaN   
2   2.0000      1      2  113781  151.5500  C22 C26        S  NaN    NaN   
3  30.0000      1      2  113781  151.5500  C22 C26        S  NaN  135.0   
4  25.0000      1      2  113781  151.5500  C22 C26        S  NaN    NaN   

                         home.dest  
0                     St Louis, MO  
1  Montreal, PQ / Chesterville, ON  
2  Montreal, PQ / Chesterville, ON  
3  Montreal, PQ / Chesterville, ON  
4  Montreal, PQ / Chesterville, ON

In [4]:
avg_age = df['age'].mean()

avg_age

29.8811345124283

In [5]:
age_sex = df['age'].groupby(df['sex'])

age_sex.mean()

sex
female    28.687071
male      30.585233
Name: age, dtype: float64

In [6]:
age_survive = df['age'].groupby(df['survived'])

age_survive.mean()

survived
0    30.545369
1    28.918228
Name: age, dtype: float64

In [7]:
age_embark = df['age'].groupby(df['embarked'])

age_embark.mean()

embarked
C    32.332154
Q    28.630000
S    29.245205
Name: age, dtype: float64

In [8]:
age_class = df['age'].groupby(df['pclass'])

age_class.mean()

pclass
1    39.159918
2    29.506705
3    24.816367
Name: age, dtype: float64

In [9]:
df['pclass'].groupby(df['pclass']).count()

pclass
1    323
2    277
3    709
Name: pclass, dtype: int64

In [10]:
fare_class = df['fare'].groupby(df['pclass'])

fare_class.mean()

pclass
1    87.508992
2    21.179196
3    13.302889
Name: fare, dtype: float64

In [11]:
fare_class_sex = df.groupby(['pclass', 'sex'])['fare']

fare_class_sex.mean()

pclass  sex   
1       female    109.412385
        male       69.888385
2       female     23.234827
        male       19.904946
3       female     15.324250
        male       12.415462
Name: fare, dtype: float64

In [14]:
embark_survive = df.groupby(['embarked', 'survived'])['pclass']

embark_survive.count()

embarked  survived
C         0           120
          1           150
Q         0            79
          1            44
S         0           610
          1           304
Name: pclass, dtype: int64

In [23]:
embark = pd.value_counts(df['embarked'])

embark

S    914
C    270
Q    123
Name: embarked, dtype: int64

In [45]:
total_embark_C = (df['embarked'] == 'C').sum()
total_survive_C = ((df['embarked'] == 'C') & (df['survived'] == 1)).sum()

percent_survive_C = (float(total_survive_C) / total_embark_C) * 100

percent_survive_C

55.555555555555557

In [46]:
total_embark_Q = (df['embarked'] == 'Q').sum()
total_survive_Q = ((df['embarked'] == 'Q') & (df['survived'] == 1)).sum()

percent_survive_Q = (float(total_survive_Q) / total_embark_Q) * 100

percent_survive_Q

35.772357723577237

In [47]:
total_embark_S = (df['embarked'] == 'S').sum()
total_survive_S = ((df['embarked'] == 'S') & (df['survived'] == 1)).sum()

percent_survive_S = (float(total_survive_S) / total_embark_S) * 100

percent_survive_S

33.260393873085334

In [49]:
embark_fare = df.groupby(['embarked'])['fare']

embark_fare.mean()

embarked
C    62.336267
Q    12.409012
S    27.418824
Name: fare, dtype: float64

In [54]:
embark_class = df.groupby(['embarked', 'pclass'])

embark_class.size()

embarked  pclass
C         1         141
          2          28
          3         101
Q         1           3
          2           7
          3         113
S         1         177
          2         242
          3         495
dtype: int64

In [58]:
total_female = df['sex'].value_counts()[1]
total_survive_female = ((df['sex'] == 'female') & (df['survived'] == 1)).sum()

percent_survive_female = (float(total_survive_female) / total_female) * 100

percent_survive_female

72.746781115879827

In [59]:
total_male = df['sex'].value_counts()[0]
total_survive_male = ((df['sex'] == 'male') & (df['survived'] == 1)).sum()

percent_survive_male = (float(total_survive_male) / total_male) * 100

percent_survive_male

19.098457888493474

In [69]:
df['pclass'].value_counts()

3    709
1    323
2    277
Name: pclass, dtype: int64

In [71]:
total_1class = df['pclass'].value_counts()[1]
total_survive_1class = ((df['pclass'] == 1) & (df['survived'] == 1)).sum()

percent_survive_1class = (float(total_survive_1class) / total_1class) * 100

percent_survive_1class

61.919504643962853

In [64]:
total_2class = df['pclass'].value_counts()[2]
total_survive_2class = ((df['pclass'] == 2) & (df['survived'] == 1)).sum()

percent_survive_2class = (float(total_survive_2class) / total_2class) * 100

percent_survive_2class

42.960288808664259

In [72]:
total_3class = df['pclass'].value_counts()[3]
total_survive_3class = ((df['pclass'] == 3) & (df['survived'] == 1)).sum()

percent_survive_3class = (float(total_survive_3class) / total_3class) * 100

percent_survive_3class

25.52891396332863

In [95]:
bins = [0, 18, 50, 100]

group_names = ['child', 'adult', 'senior']

df['ageGroup'] = pd.cut(df['age'], bins, labels=group_names)

In [96]:
df['ageGroup'].value_counts()

adult     758
child     193
senior     95
Name: ageGroup, dtype: int64

In [97]:
total_child = df['ageGroup'].value_counts()[1]
total_survive_child = ((df['ageGroup'] == 'child') & (df['survived'] == 1)).sum()

percent_survive_child = (float(total_survive_child) / total_child) * 100

percent_survive_child

49.222797927461137

In [98]:
total_adult = df['ageGroup'].value_counts()[0]
total_survive_adult = ((df['ageGroup'] == 'adult') & (df['survived'] == 1)).sum()

percent_survive_adult = (float(total_survive_adult) / total_adult) * 100

percent_survive_adult

38.786279683377309

In [99]:
total_senior = df['ageGroup'].value_counts()[2]
total_survive_senior = ((df['ageGroup'] == 'senior') & (df['survived'] == 1)).sum()

percent_survive_senior = (float(total_survive_senior) / total_senior) * 100

percent_survive_senior

40.0

In [100]:
ageGroup_class = df.groupby(['ageGroup', 'pclass'])

ageGroup_class.size()

ageGroup  pclass
adult     1         199
          2         199
          3         360
child     1          21
          2          42
          3         130
senior    1          64
          2          20
          3          11
dtype: int64

Observations:

- Average overall age is 29
- Average age for a female is 28, male is 30
- The average age of passengers that survived was 30; died is 28
- People in first class on average were older (39) and the youngest were in third class (24)
- The average fare for a female (regardless of class) was always higher than fare for a male
- Higher percentage of people that embarked at Cherbourg survived
- The average fare paid was highest for those that embarked at Cherbourg and lowest at Queenstown (heavily 3rd class passengers)
- More likely to survive if 1st class passenger; die if 3rd class passenger (61.9% 1st class, 42.9% 2nd class, 25.5% 3rd class)
- Children were most likely to survive (though chances were only about 50/50) but depending on the bins for seniors, if the bin starts at 50 then the survival rate is 40%; however, if the bin starts at 60 then the survival rate drops to 24.2%
- Most children were 3rd class passengers